frame/plasm-lockdrop/src/lib.rs

```diff
@@ -0,0 +1,662 @@
+    type DollarRate: Member + Parameter + SimpleArithmetic
+        + Copy + Default + From<u64> + Into<u128>;
+
+    // XXX: I don't known how to convert into Balance from u128 without it
+    // TODO: Should be removed
+    type BalanceConvert: From<u128> +
+        Into<<Self::Currency as Currency<<Self as frame_system::Trait>::AccountId>>::Balance>;
+
+    /// The regular events type.
+    type Event: From<Event<Self>> + Into<<Self as frame_system::Trait>::Event>;
+}
+
+/// Claim id is a hash of claim parameters.
+pub type ClaimId = H256;
+
+/// Type for enumerating claim proof votes. 
+pub type AuthorityVote = u32;
+
+/// Plasm Lockdrop parameters.
+#[cfg_attr(feature = "std", derive(PartialEq, Eq))]
+#[derive(Encode, Decode, RuntimeDebug, Clone)]
+pub enum Lockdrop {
+    /// Bitcoin lockdrop is pretty simple:
+    /// transaction sended with time-lockding opcode,
+    /// BTC token locked and could be spend some timestamp.
+    /// Duration in blocks and value in shatoshi could be derived from BTC transaction.
+    Bitcoin { public: ecdsa::Public, value: u64, duration: u64, transaction_hash: H256, },
+}
+
+impl Default for Lockdrop {
+    fn default() -> Self {
+        Lockdrop::Bitcoin {
+            public: Default::default(),
+            value: Default::default(),
+            duration: Default::default(),
+            transaction_hash: Default::default(),
+        }
+    }
+}
+
+/// Lockdrop claim request description.
+#[cfg_attr(feature = "std", derive(PartialEq, Eq))]
+#[derive(Encode, Decode, RuntimeDebug, Default, Clone)]
+pub struct Claim {
+    params:   Lockdrop,
+    approve:  AuthorityVote,
+    decline:  AuthorityVote,
+    amount:   u128,
+    complete: bool,
+}
+
+decl_event!(
+    pub enum Event<T>
+    where <T as system::Trait>::AccountId,
+          <T as Trait>::AuthorityId,
+          <T as Trait>::DollarRate,
+          Balance = BalanceOf<T>,
+    {
+        /// Lockdrop token claims requested by user
+        ClaimRequest(ClaimId),
+        /// Lockdrop token claims response by authority
+        ClaimResponse(ClaimId, AccountId, bool),
+        /// Lockdrop token claim paid
+        ClaimComplete(ClaimId, AccountId, Balance),
+        /// Dollar rate updated by oracle.
+        NewDollarRate(DollarRate),
+        /// New authority list registered
+        NewAuthorities(Vec<AuthorityId>),
+    }
+);
+
+decl_error! {
+    pub enum Error for Module<T: Trait> {
+    }
+}
+
+decl_storage! {
+    trait Store for Module<T: Trait> as Provider {
+        /// Offchain lock check requests made within this block execution.
+        Requests get(fn requests): Vec<ClaimId>;
+        /// List of lockdrop authority id's.
+        Keys get(fn keys): Vec<T::AuthorityId>;
+        /// Token claim requests.
+        Claims get(fn claims): linked_map hasher(blake2_256)
+                               ClaimId => Claim;
+        /// Lockdrop alpha parameter.
+        Alpha get(fn alpha) config(): u128;
```

Comment:
2-nd and 3-rd Lockdrop uses different `alpha`. How to change the number?

Reply:
In this PR implemented simple lockdrop, I plan to implement multi-session approach in next PR.